simvue/eco/emissions_monitor.py

```diff
@@ -0,0 +1,325 @@
+"""
+CO2 Monitor
+===========
+
+Provides an interface for estimating CO2 usage for processes on the CPU.
+"""
+
+__author__ = "Kristian Zarebski"
+__version__ = "0.1.0"
+__license__ = "MIT"
+__date__ = "2025-02-27"
+
+import datetime
+import json
+import pydantic
+import dataclasses
+import threading
+import time
+import logging
+import typing
+import psutil
+import humanfriendly
+import pathlib
+import os
+
+from simvue.eco.api_client import APIClient, CO2SignalResponse
+
+TIME_FORMAT: str = "%Y_%m_%d_%H_%M_%S"
+
+
+@dataclasses.dataclass
+class ProcessData:
+    process: psutil.Process
+    cpu_percentage: float = 0.0
+    power_usage: float = 0.0
+    total_energy: float = 0.0
+    energy_delta: float = 0.0
+    co2_emission: float = 0.0
+    co2_delta: float = 0.0
+
+
+class CO2Monitor(pydantic.BaseModel):

```

Comment:
Please reference important assumptions made here e.g. default value of thermal_design_power_per_core to documentation on CO2 emission measurement. This should be made easily accessible.

Reply:
Added to "metrics" section of docs in: https://github.com/simvue-io/docs/pull/61